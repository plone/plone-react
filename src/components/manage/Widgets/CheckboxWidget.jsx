/**
 * CheckboxWidget component.
 * @module components/manage/Widgets/CheckboxWidget
 */

import { FormFieldWrapper } from '@plone/volto/components';
import PropTypes from 'prop-types';
import React from 'react';
import { injectIntl } from 'react-intl';
import { Checkbox } from 'semantic-ui-react';

/**
 * CheckboxWidget component class.
 * @function CheckboxWidget
 * @returns {string} Markup of the component.
 */
const CheckboxWidget = ({
  id,
  title,
  required,
  description,
  error,
  value,
  onChange,
  onEdit,
  onDelete,
  intl,
  fieldSet,
  wrapped,
  isDraggable,
  isDissabled,
}) => {
  return (
    <FormFieldWrapper
      id={id}
      title={title}
      description={description}
      required={required}
      error={error}
      fieldSet={fieldSet}
      wrapped={wrapped}
      columns={1}
      draggable={isDraggable}
      onEdit={onEdit ? () => onEdit(id) : null}
      onDelete={onDelete}
      intl={intl}
      isDissabled={isDissabled}
    >
      <div className="wrapper">
        {onEdit && (
          <i aria-hidden="true" className="grey bars icon drag handle" />
        )}
        <Checkbox
          name={`field-${id}`}
          checked={value}
          disabled={isDissabled}
          onChange={(event, { checked }) => onChange(id, checked)}
          label={<label htmlFor={`field-${id}`}>{title}</label>}
        />
      </div>
    </FormFieldWrapper>
  );
};

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
CheckboxWidget.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  required: PropTypes.bool,
  error: PropTypes.arrayOf(PropTypes.string),
  value: PropTypes.bool,
  onChange: PropTypes.func,
  onEdit: PropTypes.func,
  onDelete: PropTypes.func,
  wrapped: PropTypes.bool,
  intl: PropTypes.object,
  fieldSet: PropTypes.string,
  isDraggable: PropTypes.bool,
  isDissabled: PropTypes.bool,
};

/**
 * Default properties.
 * @property {Object} defaultProps Default properties.
 * @static
 */
CheckboxWidget.defaultProps = {
  id: '',
  title: '',
  description: null,
  required: false,
  onChange: null,
  onEdit: null,
  onDelete: null,
  error: [],
  value: null,
  wrapped: false,
  intl: { formatMessage: () => {} },
  fieldSet: null,
  isDraggable: false,
  isDissabled: false,
};

export default injectIntl(CheckboxWidget);
