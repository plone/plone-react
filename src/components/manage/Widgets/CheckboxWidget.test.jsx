import React from 'react';
import { Provider } from 'react-intl-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import CheckboxWidget from './CheckboxWidget';

const mockStore = configureStore();

test('renders a checkbox widget component', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });

  const component = renderer.create(
    <Provider store={store}>
      <CheckboxWidget
        id="my-field"
        title="My field"
        description="Test"
        onChange={() => {}}
        onEdit={() => {}}
        onDelete={() => {}}
        required={true}
        error={['error1']}
        value={true}
        wrapped={false}
        intl={{ formatMessage: () => {} }}
        fieldSet={null}
        isDraggable={true}
        isDisabled={false}
      />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
