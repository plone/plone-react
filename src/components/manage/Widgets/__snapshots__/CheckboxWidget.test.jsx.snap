// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a checkbox widget component 1`] = `
Array [
  <div
    className="wrapper"
  >
    <div
      className="toolbar"
    >
      <button
        aria-label="Edit"
        className="item ui noborder button"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="blue write square large icon"
          onClick={[Function]}
        />
      </button>
      <button
        aria-label="Delete"
        className="item ui noborder button"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="red close large icon"
          onClick={[Function]}
        />
      </button>
    </div>
    <i
      aria-hidden="true"
      className="grey bars icon drag handle"
    />
    <div
      className="ui checked checkbox"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={true}
        className="hidden"
        disabled={false}
        name="field-my-field"
        readOnly={true}
        tabIndex={0}
        type="checkbox"
      />
      <label
        htmlFor="field-my-field"
      >
        My field
      </label>
    </div>
  </div>,
  <div
    className="ui red pointing basic label"
    onClick={[Function]}
  >
    error1
  </div>,
]
`;
